sdk_extensions/logging/support/src/main/java/io/opentelemetry/sdk/logging/LogProcessor.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.logging;
+
+import io.opentelemetry.sdk.logging.data.LogRecord;
+import io.opentelemetry.sdk.trace.TracerSdkProvider;
+
+public interface LogProcessor {
+
+  void addLogRecord(LogRecord record);
+
+  /**
+   * Called when {@link TracerSdkProvider#shutdown()} is called.
+   *
+   * <p>Implementations must ensure that all span events are processed before returning.
+   */
+  void shutdown();
```

Comment:
We're returning `CompletableResultCode` for shutdown / forceFlush now, can we update these?